### PR TITLE
fix(avoidance): turn off blinker when the ego return original lane

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description

Previously, avoidance module continued to turn on blinker until the ego returned original path completely. Sometimes, the module keeped turn signal even when the ego stopped in front of intersection due to red traffic light. However, the signal maybe causes misunderstanding that the ego is going to turn left or right at the intersection.

In this PR, I fixed turn signal calculation logic in order to disable turn signal if the ego stopped in original lane.

https://github.com/autowarefoundation/autoware.universe/assets/44889564/97ca3497-0107-40a0-bdd4-a4c1644140bc

## Tests performed

- [x] [PASS TIER IV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/b7445d9d-8e98-58c8-abde-a9e07c1c3a01?project_id=prd_jt)
## Effects on system behavior

Nothing.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_avoidance_module/src/utils.cpp [hunk 1]
```diff
@@ -250,16 +250,37 @@ void pushUniqueVector(T & base_vector, const T & additional_vector)
   base_vector.insert(base_vector.end(), additional_vector.begin(), additional_vector.end());
 }
 
+bool isAvoidShift(const double start_shift_length, const double end_shift_length)
+{
+  constexpr double THRESHOLD = 0.1;
```
Comment (Owen-Liuyuxuan) [inline planning/behavior_path_avoidance_module/src/utils.cpp:255]:
```
bool isAvoidShift(const double start_shift_length, const double end_shift_length, const double THRESHOLD = 0.1)
{
  return std::abs(start_shift_length) < THRESHOLD && std::abs(end_shift_length) > THRESHOLD;
}
```

Personally, I would wonder if it is helpful to make threshold a "default" argument for these functions?

File: planning/behavior_path_avoidance_module/src/utils.cpp [hunk 2]
```diff
@@ -314,6 +331,27 @@ bool existShiftSideLane(
...
+    return false;
+  }
+
+  for (const auto & lane : original_lanes) {
+    if (within(to2D(toLaneletPoint(ego_pos)), lane.polygon2d().basicPolygon())) {
+      return true;
+    }
+  }
+
+  return false;
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_avoidance_module/src/utils.cpp:352]: personally I would suggest `std::any_of` for this part.

File: planning/behavior_path_avoidance_module/src/utils.cpp (hunk 1 above)
Comment (satoshi-ota) [reply planning/behavior_path_avoidance_module/src/utils.cpp]: Fixed in https://github.com/autowarefoundation/autoware.universe/pull/6300/commits/e677db4309abc6d15f879b0fded9cdf96247ef81.

File: planning/behavior_path_avoidance_module/src/utils.cpp [hunk 3]
```diff
@@ -2286,7 +2324,7 @@ double calcDistanceToReturnDeadLine(
   return distance_to_return_dead_line;
 }
 
-TurnSignalInfo calcTurnSignalInfo(
+std::pair<TurnSignalInfo, bool> calcTurnSignalInfo(
```
[the remark this answers is not in the document]
Comment (satoshi-ota) [reply planning/behavior_path_avoidance_module/src/utils.cpp:2357]: Fixed in https://github.com/autowarefoundation/autoware.universe/commit/e677db4309abc6d15f879b0fded9cdf96247ef81.

**Pull request merged**